presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotAggregationProjectConverter.java

```diff
@@ -217,17 +231,46 @@ private PinotExpression handleFunction(
             CallExpression function,
             Map<VariableReferenceExpression, PinotQueryGeneratorContext.Selection> context)
     {
-        switch (function.getDisplayName().toLowerCase(ENGLISH)) {
+        String functionName = function.getDisplayName().toLowerCase(ENGLISH);
+        switch (functionName) {
             case "date_trunc":
                 boolean useDateTruncation = PinotSessionProperties.isUseDateTruncation(session);
                 return useDateTruncation ?
                         handleDateTruncationViaDateTruncation(function, context) :
                         handleDateTruncationViaDateTimeConvert(function, context);
+            case "array_max":
+            case "array_min":
+                String pinotArrayFunctionName = PRESTO_TO_PINOT_ARRAY_AGGREGATIONS.get(functionName);
+                requireNonNull(pinotArrayFunctionName, "Converted Pinot array function is null for - " + functionName);
+                return derived(String.format("%s(%s)", pinotArrayFunctionName,
+                        function.getArguments().get(0).accept(this, context).getDefinition()));
+            // array_sum and array_reduce are translated to a reduce function with lambda functions, so we pass in
+            // this arrayVariableHint to help determine which array function it is.
+            case "reduce":
```

Comment:
I could not see any unit tests for this reduce codepath. Are they indeed there ?

Reply:
I'm just too lazy to write reduce functions for average and sum :) 

Added those tests.